utils/bids/bids_file_locator.py

```diff
@@ -0,0 +1,248 @@
+                            dwis = [ dwi_dict['dwiFile'][i] for i in idxs if dwi_dict['direction'][i] == PEdirNum ]
+                            assert len(dwis) <= 2
+                            dwiname = "Diffusion" + "_dir-" + str(dirnum) + "_" + numrun + "_corr_" + str(PEdirNum)
+                            if "j" in PEdirNum:
+                                PEdir = 2
+                            elif "i" in PEdirNum:
+                                PEdir = 1
+                            else:
+                                RuntimeError("Phase encoding direction not specified for diffusion data.")
+                            pos = "EMPTY"
+                            neg = "EMPTY"
+                            gdcoeffs = "None"
+                            for dwi in dwis:
+                                if "-" in layout.get_metadata(dwi)["PhaseEncodingDirection"]:
+                                    neg = dwi
+                                else:
+                                    pos = dwi
+
+                                echospacing = layout.get_metadata(pos)["EffectiveEchoSpacing"] * 1000
+    #TODO make sure there are not naming collisions in the config dict such that parameters (e.g., echospacing
+    #is overwritten from structural setup to func to diff
+                    self.config.update({
+                        "dwiname":dwiname,
+                        "PEdir":PEdir,
+                        "pos": pos,
+                        "neg":neg,
+                        "echospacing": echospacing
+                    })
+
+    def grab_BIDS_data(self):
+        """
+        Download and validate the BIDS directory structure from curated data.
+        This method must be included since the files are copied up to Docker and the
+        file organization will need to follow the BIDS curation/specification.
+        Args:
+            gtk_context (GearToolkitContext): gear information
+        Returns:
+            msg (str): Status of BIDS download (and validation)
+        """
+        log.info("Checking BIDS")
+        config = self.gtk_context.config
+        try:
+            log.info('Downloading BIDS')
+            self.gtk_context.download_session_bids(target_dir=self.gtk_context.work_dir / "bids")
+            msg = "BIDS downloaded, but validation was not required."
+        except Exception as e:
+            log.error(
+                f'Could not download BIDS data from {self.gtk_context.work_dir / "bids"}'
+            )
+            log.exception(e)
+            msg = "Errors in BIDS download."
+
+        if config["gear-run-bids-validation"] and config["gear-abort-on-bids-error"]:
+            # Pass the BIDS dir to the commandline BIDS validator that virtually all BIDS apps
+            # use and check for errors.
+            # Error logging handled within validate_bids.
+
+            # Need to convert from cli to python bids_validator (npm unstable solution)
```

Comment:
Last I looked, the python BIDS validator only validates individual files while the regular one goes through everything.  I'm not sure that makes a difference.  In any event, this can be dealt with later.

Reply:
On the todo list. Thank you.